corehq/apps/registration/forms.py

```diff
@@ -632,9 +644,20 @@ def clean(self):
             raise forms.ValidationError(error)
         return cleaned_data
 
-    def _initialize_tableau_fields(self, data, domain):
+    def _initialize_tableau_fields(self, data, domain, invitation=None):
         self.tableau_form = BaseTableauUserForm(data, domain=domain)
         self.fields['tableau_group_indices'] = self.tableau_form.fields["groups"]
         self.fields['tableau_group_indices'].label = _('Tableau Groups')
         self.fields['tableau_role'] = self.tableau_form.fields['role']
         self.fields['tableau_role'].label = _('Tableau Role')
+        if invitation:
+            initial_groups_indicies = []
+            for group_index in invitation.tableau_group_ids:
+                for i, group in enumerate(self.tableau_form.allowed_tableau_groups):
+                    if group_index == group.id:
+                        initial_groups_indicies.append(i)
+            self.fields['tableau_group_indices'].initial = initial_groups_indicies
+            try:
+                self.fields['tableau_role'].initial = TableauUser.Roles(invitation.tableau_role)
```

Comment:
why does this need to be in a try block? Also, can't it just be 

`self.fields['tableau_role'].initial= invitation.tableau_role`?

Reply:
That's what I thought too but `invitation.tableau_role` is saved as a string but I need to set the initial value equal to one of choices for this field, which is populated from `TableauUser.Roles.choices`. I'm not sure what the exact variable type these choices are (see Django TextChoices) but its not just a string. I put it in a try block for safety because trying to get a role that doesn't exist in roles.choices will throw a valueerror.